x-pack/plugins/observability/server/domain/services/date_range.ts

```diff
@@ -23,7 +23,12 @@ export const toDateRange = (timeWindow: TimeWindow, currentDate: Date = new Date
       throw new Error('Cannot compute date range with future starting time');
     }
 
-    const from = startTime.clone().add(differenceInUnit, unit);
+    const from = startTime
+      .clone()
+      .add(
+        Math.floor(differenceInUnit / timeWindow.duration.value) * timeWindow.duration.value,
+        unit
+      );
```

Comment:
🐛 This was buggy for duration with non-unitary duration values, e.g. `2w`, `30d`, `3Q`, etc...